Sources/ScipioKit/Producer/PIF/PIFGenerator.swift

```diff
@@ -1,308 +1,208 @@
         switch frameworkType {
         case .dynamic, .mergeable:
-            settings[.MACH_O_TYPE] = "mh_dylib"
+            configuration.buildSettings["MACH_O_TYPE"] = "mh_dylib"
         case .static:
-            settings[.MACH_O_TYPE] = "staticlib"
+            configuration.buildSettings["MACH_O_TYPE"] = "staticlib"
         }
 
-        settings[.LIBRARY_SEARCH_PATHS, default: ["$(inherited)"]]
+        configuration.buildSettings["LIBRARY_SEARCH_PATHS"]
             .append("\(toolchainLibDir.pathString)/swift/\(sdk.settingValue)")
 
         // Enable to emit swiftinterface
         if buildOptions.enableLibraryEvolution {
-            settings[.OTHER_SWIFT_FLAGS, default: ["$(inherited)"]]
+            configuration.buildSettings["OTHER_SWIFT_FLAGS"]
                 .append("-enable-library-evolution")
-            settings[.SWIFT_EMIT_MODULE_INTERFACE] = "YES"
+            configuration.buildSettings["SWIFT_EMIT_MODULE_INTERFACE"] = "YES"
         }
-        settings[.SWIFT_INSTALL_OBJC_HEADER] = "YES"
+        configuration.buildSettings["SWIFT_INSTALL_OBJC_HEADER"] = "YES"
 
         if frameworkType == .mergeable {
-            settings[.OTHER_LDFLAGS, default: ["$(inherited)"]]
+            configuration.buildSettings["OTHER_LDFLAGS"]
                 .append("-Wl,-make_mergeable")
         }
 
-        appendExtraFlagsByBuildOptionsMatrix(to: &settings)
+        appendExtraFlagsByBuildOptionsMatrix(to: &configuration, target: target)
 
         // Original PIFBuilder implementation of SwiftPM generates modulemap for Swift target
         // That modulemap refer a bridging header by a relative path
         // However, this PIFGenerator modified productType to framework.
         // So a bridging header will be generated in frameworks bundle even if `SWIFT_OBJC_INTERFACE_HEADER_DIR` was specified.
         // So it's need to replace `MODULEMAP_FILE_CONTENTS` to an absolute path.
-        if let swiftTarget = resolvedTarget.underlying as? ScipioSwiftModule {
-            // Bridging Headers will be generated inside generated frameworks
-            let productsDirectory = descriptionPackage.productsDirectory(
-                buildConfiguration: buildOptions.buildConfiguration,
-                sdk: sdk
-            )
-            let bridgingHeaderFullPath = productsDirectory.appending(
-                components: ["\(swiftTarget.c99name).framework", "Headers", "\(swiftTarget.name)-Swift.h"]
-            )
-
-            settings[.MODULEMAP_FILE_CONTENTS] = """
-                module \(swiftTarget.c99name) {
-                    header "\(bridgingHeaderFullPath.pathString)"
-                    export *
-                }
-                """
+        if case .string(let moduleMapFileContents) = configuration.buildSettings["MODULEMAP_FILE_CONTENTS"],
+           moduleMapFileContents.contains("\(target.name)-Swift.h") {
+            resolveModuleMapPath(of: target, configuration: &configuration, sdk: sdk)
```

Comment:
```suggestion
        if case .string(let moduleMapFileContents) = configuration.buildSettings["MODULEMAP_FILE_CONTENTS"],
           moduleMapFileContents.contains("\(target.name)-Swift.h")
        {
            resolveModuleMapPath(of: target, configuration: &configuration, sdk: sdk)
```

would be more readable (for me).